Dockerfile

```
@@ -0,0 +1,41 @@
+FROM python:3.9-bullseye  as build
+
+RUN apt update \
+ && apt install -y python3-virtualenv
+
+WORKDIR /opt
+
+COPY . moonraker-telegram-bot/
+
+RUN virtualenv -p /usr/bin/python3 --system-site-packages venv \
+ && venv/bin/pip install --no-cache-dir -r moonraker-telegram-bot/scripts/requirements.txt
+
+FROM python:3.9-slim-bullseye as run
+
+RUN apt update \
+ && apt install -y \
+      python3-cryptography \
+      python3-gevent \
+      python3-opencv \
+      x264 \
+      libx264-dev \
+      libwebp-dev \
+      && apt clean \
+      && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /opt
+
+RUN groupadd moonraker-telegram-bot --gid 1000 \
+ && useradd moonraker-telegram-bot --uid 1000 --gid moonraker-telegram-bot
+
+COPY --chown=moonraker-telegram-bot:moonraker-telegram-bot --from=build /opt/moonraker-telegram-bot ./moonraker-telegram-bot
+COPY --chown=moonraker-telegram-bot:moonraker-telegram-bot --from=build /opt/venv ./venv
+
+RUN mkdir -p printer_data/logs printer_data/config timelapse timelapse_finished \
+ && chown -R moonraker-telegram-bot:moonraker-telegram-bot /opt/*
```

When not using a multistage build, the venv does not have to be copied anymore and the copy command for the codebase needs to reference the build context.

Other then that, using chown for files created in a different layer, creates a new layer that takes up as much space as the one before, doubling the space used in the image. 

Merging the user/directory creation with the permission change creates only a single layer. The venv does not need to be owned by the user.

```suggestion
RUN groupadd moonraker-telegram-bot --gid 1000 \
 && useradd moonraker-telegram-bot --uid 1000 --gid moonraker-telegram-bot \
 && mkdir -p printer_data/logs printer_data/config timelapse timelapse_finished \
 && chown -R moonraker-telegram-bot:moonraker-telegram-bot /opt/*

COPY --chown=moonraker-telegram-bot:moonraker-telegram-bot . ./moonraker-telegram-bot
RUN python3 -m venv venv \
 && venv/bin/pip install -r moonraker-telegram-bot/scripts/requirements.txt
```